### ISSUE TITLE
Add some retry around launching browsers

### ISSUE THREAD

**Issue by natebosch** [dart-lang/test#1578]
https://github.com/dart-lang/test/blob/d6329d676c7578c984a5397a6cc4002757c0b21e/pkgs/test/lib/src/runner/browser/browser_manager.dart#L129-L133

We seem to hit timeouts much more often on github actions.

**Comment by natebosch** [comment]
@grouma - does `package:browser_launcher` already have retry?

https://github.com/dart-lang/test/issues/1156

**Comment by grouma** [comment]
I don't think so: https://github.com/dart-lang/browser_launcher/blob/master/lib/src/chrome.dart
